### PR TITLE
Issue 1769: MapEditor New Map Cancel button

### LINKED ISSUE #1769: 0.47.5-SNAPSHOT - Board Editor RFE: Add Close Button to New Board Dialog
Currently, if you open the new board dialog it will always regenerate the map. That could be undesirable if you misclick it.

### PR DESCRIPTION
**Pull request by SJuliez**
Resolves #1769

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/RandomMapDialog.java [hunk 1]
```diff
@@ -414,6 +428,17 @@ private boolean doApply() {
         MAP_SETTINGS_OBSERVER.updateMapSettings(newMapSettings);
         return true;
     }
+    
+    public boolean activateDialog() {
+    	userCancel = false;
+    	setVisible(true);
+    	return userCancel;
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/RandomMapDialog.java]: This seems kind of pointless - you're setting userCancel to false then returning it without any possibility of changing it.

File: megamek/src/megamek/client/ui/swing/BoardEditor.java [hunk 2]
```diff
@@ -1330,18 +1330,21 @@ private void updateWhenSelected() {
...
-        rmd.setVisible(true);
-        board = BoardUtilities.generateRandom(mapSettings);
-        game.setBoard(board);
-        curfile = null;
-        frame.setTitle(Messages.getString("BoardEditor.title")); //$NON-NLS-1$
-        menuBar.setBoard(true);
-        bvc.doLayout();
+    	RandomMapDialog rmd = new RandomMapDialog(frame, this, null, mapSettings);
+    	boolean userCancel = rmd.activateDialog();
+    	if (!userCancel) {
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/BoardEditor.java]:
```suggestion
    	if (!rmd.activateDialog()) {
```

File: megamek/src/megamek/client/ui/swing/BoardEditor.java [hunk 3]
```diff
@@ -1330,18 +1330,21 @@ private void updateWhenSelected() {
...
+    		menuBar.setBoard(true);
+    		bvc.doLayout();
+    		resetUndo();
+    	}
     }
-    
+
     public void boardResize() {
-        ResizeMapDialog emd = new ResizeMapDialog(frame, this, null, mapSettings);
+    	ResizeMapDialog emd = new ResizeMapDialog(frame, this, null, mapSettings);
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/BoardEditor.java]: Formatting on this line is weird.

File: megamek/src/megamek/client/ui/swing/RandomMapDialog.java (hunk 1 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/RandomMapDialog.java]: This is a modal dialog. setVisible does not return until the dialog is closed.

File: megamek/src/megamek/client/ui/swing/BoardEditor.java (hunk 2 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/BoardEditor.java]: I'd rather waste that one line for improved readability.

File: megamek/src/megamek/client/ui/swing/BoardEditor.java (hunk 3 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/BoardEditor.java]: Didn't actively change that line. But I noticed that my formatter was still using tabs which Ive changed now. I'd rather leave this for now and replace tabs across some files at another time.

**Pull request merged**